### ISSUE TITLE
Unexpectedly weird errors while compiling a project with opencv-rust

### ISSUE THREAD

**Issue by timleg002** [twistedfall/opencv-rust#524]
[12.12.2023 OPENCV-RUST LOG.txt](https://github.com/twistedfall/opencv-rust/files/13648837/12.12.2023.OPENCV-RUST.LOG.txt)
1. Operating system

    Windows 11 Pro 21H2 

3. The way you installed OpenCV: package, official binary distribution, manual compilation, etc.

    Via chocolatey

5. OpenCV version

    `4.8.1`

7. rustc version (`rustc --version`)
    
    `rustc 1.76.0-nightly (21cce21d8 2023-12-11)`

I have attached the full log, but in short: very weird errors. I haven't been able to find anything *at all* about this. This error occurs with opencv bindgen, failing to run the bindings generator for every opencv module. This line keeps repeating (with some variations): `c:/MinGW64/x86_64-w64-mingw32/include\stdlib.h:388:47: error: expected ';' after top level declarator`. It seems as if I was using some weird compilator? I don't know. I tried both the msvc and the gnu target triple (for x64 Windows), both give the same error.

**Comment by twistedfall** [comment]
I think you haven't in fact attached the full build log unless I'm missing something

**Comment by timleg002** [comment]
> I think you haven't in fact attached the full build log unless I'm missing something

You're right - I uploaded the file, but for some reason it wasn't there. Thank you for the fast reply - it's attached now!

**Comment by twistedfall** [comment]
Looks like the stdlib headers are not processed correctly. I'm not sure how to help here to be honest and I've never tested build with mingw. There is also this warning that's very suspicious in the logs:
```
warning: "__STRICT_ANSI__ seems to have been undefined; this is not supported"
```

**Comment by timleg002** [comment]
I also tried building with MSVC, but got the same errors (with the stdlib headers). It all feels like I made a mistake configuring all of this, but I retraced all of my steps and don't think I found anything wrong.

**Comment by icedcoffeeee** [comment]
Sorry for bumping, currently facing the same issue. Any work arounds?

**Comment by twistedfall** [comment]
Are you using mingw on Windows too?

**Comment by icedcoffeeee** [comment]
Yes, I am. My errors are equivalent to OP.

**Comment by sampaioletti** [comment]
I had this and for me I had to remove Strawberry Perl that I had on my machine for another project...YMMV but that made the errors go away for me. Maybe its unrelated and I was just very lucky (:
